src/Common/src/System/Marvin.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
It looks like CoreRT was not testing this directly, only through String.GetHashCode.

Could you please add a few tests? They would go in corefx\src\Common\tests\System and corefx\src\Common\tests\Common.TEsts.csproj. You run them with the usual `msbuild /t:buildandtest` in the csproj folder.

Perhaps there are some canonical marvin inputs/outputs somewhere.

Reply:
@AtsushiKan did you preserve any of the test data from when you ported this into corert?

Reply:
Test data is only useful if you can force the seed to a value of your own choosing. If it helps, though, here's the data I used: https://github.com/AtsushiKan/corert/tree/marvin-wip/src/Marvin

Reply:
That's an awesome body of test cases. I would settle for 30-50 adding into corefx randomly selected cases, it should be pretty hard to break the code and still have all of them pass.

Reply:
@danmosemsft Do you mean to use these specific test data values? That would be again going indirectly through `String.GetHashCode` as these are string inputs. I can however just generate a few `byte[]` and calculate match their `Marvin` 32/64 bit hashes, but I would be doing that from the current implementation (which only tests the existing implementation). What do you prefer?

Reply:
@i3arnon - The test data are byte arrays - I just used hex strings to represent them in source (for succinctness - no other reason.) A `HexStringToByteArray()` utility method (not shown - but we've probably got several variants in corefx already under the `Common\Tests` subdirectory) converts them to byte arrays at runtime.

Reply:
@i3arnon I may be misunderstanding, but it seems that @AtsushiKan was ingesting these as byte[] https://github.com/AtsushiKan/corert/blob/marvin-wip/src/Marvin/MarvinTest.cs#L18 and then hashing and comparing to the expected long result. I don't think there is any string hashing there?

The purpose of doing this would be some basic protection against someone breaking the implementation you're pasting in. 

The whole body of tests would be too big for this repo at 5MB, that would need to go into our test-data repo, I don't think you need to go to that trouble. We may expose Marvin as a public API, that would be a good time to do that.

Reply:
You're right, I missed it. I went ahead and added a test with some of this test data.